packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -4546,11 +4564,13 @@ export class ContainerRuntime
 	private reSubmitBatch(
 		batch: PendingMessageResubmitData[],
 		batchId: BatchId,
-		staged: boolean,
+		{ staged, squash }: { staged: boolean; squash: boolean },
 	): void {
 		this.batchRunner.run(() => {
 			for (const message of batch) {
-				this.reSubmit(message);
+				// Note: once squashing is widely supported, squashing even for orderSequentially should be fine.
+				// See remarks on orderSequentiallyRunner for more details.
+				this.reSubmit(message, this.orderSequentiallyRunner.running ? false : squash);
```

Comment:
i assume some of this will change once squash is op in by the caller?

Reply:
yep, it'll go away with that change.